File: kotlin-remix-run-router/karakum/inheritanceModifiers/modifyInterfaceInheritance.js
import ts from "typescript"

export default (node) => {
    if (
        ts.isInterfaceDeclaration(node)
        && (node.name.text === "AgnosticRouteMatch"
            || node.name.text === "AgnosticPatchRoutesOnMissFunction")
    ) {
        return ""
    }

    if (
        ts.isInterfaceDeclaration(node)
        || ts.isTypeAliasDeclaration(node)
        || ts.isTypeLiteralNode(node)
        || ts.isIntersectionTypeNode(node)
    ) {
        return "sealed"
    }

    return null
}

File: kotlin-typescript/karakum/src/jsMain/kotlin/typescript/karakum/inheritanceModifiers/modifyInterfaceInheritance.kt
package typescript.karakum.inheritanceModifiers

import arrow.core.raise.nullable
import io.github.sgrishchenko.karakum.extension.InheritanceModifierContext
import typescript.Node
import typescript.isInterfaceDeclaration
import typescript.isIntersectionTypeNode
import typescript.isTypeAliasDeclaration
import typescript.isTypeLiteralNode

fun modifyInterfaceInheritance(node: Node, context: InheritanceModifierContext) = nullable {
    ensure(
        isInterfaceDeclaration(node)
                || isTypeAliasDeclaration(node)
                || isTypeLiteralNode(node)
                || isIntersectionTypeNode(node)
    )

    "sealed"
}
